From da3973dedc9cf4d0ac7ada41046bd2199847884f Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Thu, 21 Sep 2017 12:53:55 -0400
Subject: [PATCH] OS-6311 test-delete.js: prefer node libuuid binding over
 external utility Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
 Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon
 <josh@wilsdon.ca>

---
 src/vm/tests/test-delete.js | 51 ++++++++-----------------------------
 1 file changed, 11 insertions(+), 40 deletions(-)

diff --git a/src/vm/tests/test-delete.js b/src/vm/tests/test-delete.js
index 055e1457..ee472d6c 100644
--- a/src/vm/tests/test-delete.js
+++ b/src/vm/tests/test-delete.js
@@ -1,12 +1,11 @@
-// Copyright 2014 Joyent, Inc.  All rights reserved.
+// Copyright 2017 Joyent, Inc.
 //
 // These tests ensure that delete behaves correctly.
 //
 
 var async = require('/usr/node/node_modules/async');
-var execFile = require('child_process').execFile;
+var libuuid = require('/usr/node/node_modules/uuid');
 var VM = require('/usr/vm/node_modules/VM');
-var vmtest = require('../common/vmtest.js');
 
 // this puts test stuff in global, so we need to tell jsl about that:
 /* jsl:import ../node_modules/nodeunit-plus/index.js */
@@ -14,53 +13,25 @@ require('nodeunit-plus');
 
 VM.loglevel = 'DEBUG';
 
-// blatantly copied from VM.js
-function rtrim(str, chars)
-{
-    chars = chars || '\\s';
-    str = str || '';
-    return str.replace(new RegExp('[' + chars + ']+$', 'g'), '');
-}
-
-function getUUID(callback)
-{
-    var cmd = '/usr/bin/uuid';
-
-    execFile(cmd, ['-v', '4'], function (error, stdout, stderr) {
-        if (error) {
-            console.error('uuid exited non-zero (' + error.code + ')');
-        }
-        callback(error, rtrim(stdout));
-    });
-}
-
 // w/ OS-2284 we sometimes blew up deleting a non-existent VM, test that we
 // haven't regressed.
 test('test deleting nonexistent VM', function(t) {
-    var i;
-
-    i = 0;
+    var i = 0;
     async.whilst(
         function () {
             return i < 50;
         },
         function (callback) {
-            getUUID(function (err, uuid) {
-                i++;
-                t.ok(!err, 'no error getting uuid: ' + (err ? err.message : 'ok'));
-                if (!err) {
-                    t.ok(uuid, 'uuid is: ' + uuid);
-                    VM.delete(uuid, {}, function (err) {
-                        if (err && err.message.match(/No such zone configured/)) {
-                            t.ok(true, 'zone ' + uuid + ' already does not exist, skipping');
-                        } else {
-                            t.ok(!err, 'deleted VM: ' + (err ? err.message : 'success'));
-                        }
-                        callback();
-                    });
+            i++;
+            var uuid = libuuid.create();
+            t.ok(uuid, 'uuid is: ' + uuid);
+            VM.delete(uuid, {}, function (err) {
+                if (err && err.message.match(/No such zone configured/)) {
+                    t.ok(true, 'zone ' + uuid + ' already does not exist, skipping');
                 } else {
-                    callback();
+                    t.ok(!err, 'deleted VM: ' + (err ? err.message : 'success'));
                 }
+                callback();
             });
         }, function (err, cb) {
             t.end();
-- 
2.21.0

